### PR TITLE
[IMP] sale: mention documents in quotation mail

### PR DESCRIPTION
**Pull request by livaios**
Description of the issue/feature this PR addresses:
Currently we have product documents that are available for the customer in the portal but the email does not mention it so the customer may not be aware if he does not navigate to the portal.

Desired behavior after PR is merged:
Added the list of documents links to the portal at the end of the quotation and order confirmation email if there are documents to share at quote

task-3637434

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale_order.py [hunk 1]
```diff
@@ -1849,6 +1849,9 @@ def _filter_product_documents(self, documents):
                 or (self.state == 'sale' and document.attached_on == 'sale_order')
         )
 
+    def get_product_documents_names(self):
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:1852]:
```suggestion
    def _get_product_documents_names(self):
```

File: addons/sale/models/sale_order.py [hunk 2]
```diff
@@ -1849,6 +1849,9 @@ def _filter_product_documents(self, documents):
...
+        return self._get_product_documents().mapped('ir_attachment_id').mapped('name')
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:1853]:
```suggestion
        return self._get_product_documents().mapped('ir_attachment_id.name')
```

File: addons/sale/models/sale_order.py (hunk 1 above)
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:1852]: Add a `self.ensure_one()`, don't rely on the one in `_get_product_documents` ;)

**Review by Demesmaeker — CHANGES_REQUESTED** [review]
Hello :duck: 
Nice one! =D

**Review by Demesmaeker — APPROVED** [review]

**Review by Demesmaeker — APPROVED** [review]

File: addons/sale/data/mail_template_data.xml [hunk 3]
```diff
@@ -29,7 +29,28 @@
...
+        <t t-set="documents" t-value="object._get_product_documents()"/>
+        <t t-if="documents">
+            <br/>Here 
+            <t t-if="len(documents)>1">
+                are some additional documents
+            </t>
+            <t t-else="">
+                is an additional document
+            </t>
+            that may interest you:
```
Comment (Feyensv) [inline addons/sale/data/mail_template_data.xml:42]:
```suggestion
            <br/> 
            <t t-if="len(documents)>1">
                Here are some additional documents that may interest you:
            </t>
            <t t-else="">
                Here is an additional document that may interest you:
            </t>
```
I know the existing content of the template is a bad example of it, but to ease the work of translators, we should avoid splitting sentences as much as possible.  Depending on the language, the place of words vary a lot and the ordering you did in english may be the opposite in another language.

For mail templates, it's less an issue as the whole template is considered as one 'block' to be translated, but in some other cases, e.g. templates used on website, 'html' content is parsed to extract translatable strings.  This sometimes leads to situations where the translators only get part of sentences, if not one word alone, and it's difficult to understand the meaning of that part, and/or translate it rightly in their own language with the information/restrictions they have.

**Review by Feyensv — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by Feyensv** [comment]
Thanks for the changes :)

**Comment by david-fleischmann** [comment]
@livaios this is a real good one! I always felt like "there is missing something here".

**Pull request closed without merging**